Deck.cpp

```diff
@@ -0,0 +1,72 @@
+#include "Deck.h"
+
+void Deck::createDeckandShuffle()
+{
+  for(int i{0}; i < m_deck.size(); i++)
+  {
+    m_deck.at(i).rank = static_cast<Card::Rank>(i%13);
+    m_deck.at(i).suit = static_cast<Card::Suit>(i/13);
+  }
+  static std::mt19937 mt{static_cast<std::mt19937::result_type>(std::time(nullptr))};
+  std::shuffle(m_deck.begin(), m_deck.end(), mt);
+}
+
+int Deck::getCardValue()
+{
+    int cardValue{};
+    Card::Rank rankForComparison{};
+    if(std::find(m_deck.begin(), m_deck.end(), m_deck[m_index]) == std::end(m_deck))
```

Comment:
Hmm. That is good context to have that I otherwise would not have ever thought of. Alright, so curly braces it is.

And here I thought I was being clever by omitting the curly braces. Also, Sad